include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -216,9 +175,80 @@ class aabbSIMD<core::vector2df_SIMD> : public aabbSIMDBase<aabbSIMD, core::vecto
 
 };
 
+template<typename VECTOR_TYPE>
+class aabb2dSIMDInt : public aabbSIMDBase<aabb2dSIMDInt, VECTOR_TYPE>
+{
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value,
+		"assertion failed: cannot use this type");
+
+public:
+	inline aabb2dSIMDInt()
+	{
+		if constexpr (std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value)
+			internalBoxRepresentation = core::vector4di32_SIMD(INT_MIN);
+
+		if constexpr (std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value)
+			internalBoxRepresentation = core::vector4du32_SIMD(0u);
+	}
+
+	inline aabb2dSIMDInt(const VECTOR_TYPE& _box)
+		//:inernalboxRepresentation(_box) - doesn't compile
+	{
+		this->internalBoxRepresentation = _box;
+	}
+
+	inline void addBox(const VECTOR_TYPE& other)
+	{
+		//TODO
+	}
+
+	inline core::vector2di32_SIMD getMinEdge() const
+	{
+		core::vector2di32_SIMD minEdge = internalBoxRepresentation;
+		minEdge.makeSafe2D();
+
+		return minEdge;
+	}
+
+	inline core::vector2di32_SIMD getMaxEdge() const
+	{
+		core::vector2di32_SIMD maxEdge = internalBoxRepresentation.zwzw();
+		maxEdge.makeSafe2D();
+
+		return maxEdge;
+	} 
+
+	inline bool isPointInside(const VECTOR_TYPE& p) const
+	{
+		//need operator< and operator>
+		//but implementation will be the same for both core::vector2di_SIMD and core::vector2du_SIMD
```

Comment:
great plan, remember to use `_mm_blend_*` from SSE4.2 (the same can be used to implement `core::mix` compatibile with GLSL that operates on a bvec)

Reply:
actually if you did
```
all(unpacklo(internalBoxRepresentation,p) < mix(p,internalBoxRepresentation,bvec4(false,false,true,true)))
// all(  (i.min,p.xy) < (p.xy,i.max)  )
```
then you could get away just with `operator<`